app/src/main/java/com/example/sharingang/AccountFragment.kt

```diff
@@ -97,4 +145,78 @@ class AccountFragment : Fragment() {
             binding.accountStatus.text = getString(R.string.text_logged_out)
         }
     }
+
+    private fun restorePreferences(binding: FragmentAccountBinding) {
+        val sharedPreferences = context?.getSharedPreferences(
+            getString(R.string.preference_user_info),
+            Context.MODE_PRIVATE
+        )
+        if (sharedPreferences != null) {
+            if(sharedPreferences.getString(getString(R.string.key_account_name), "").equals("")) {
+                updateUI(AccountStatus.LOGGED_OUT, binding)
+            }
+            else {
+                "Status: Logged in as \n${sharedPreferences.getString(
+                    getString(R.string.key_account_name), "")}"
+                    .also { binding.accountStatus.text = it }
+                updateUI(AccountStatus.LOGGED_IN, binding)
+            }
+        }
+        else {
+            updateUI(AccountStatus.LOGGED_OUT, binding)
+        }
+    }
+
+    private fun firebaseAuthWithGoogle(idToken: String, binding: FragmentAccountBinding) {
+        val credential = GoogleAuthProvider.getCredential(idToken, null)
+        auth.signInWithCredential(credential)
+            .addOnCompleteListener(requireActivity()) { task ->
+                if (task.isSuccessful) {
+                    val user = auth.currentUser
+                    lifecycleScope.launch(Dispatchers.IO) {
+                        userRepository.add(
+                            User(
+                                id = user?.uid,
+                                name = user?.displayName!!,
+                                profilePicture = user.photoUrl!!.toString()
+                            )
+                        )
+                    }
+                    editor.putString(getString(R.string.account_firebase_uid), user?.uid)
```

Comment:
```suggestion
                if (task.isSuccessful && auth.currentUser != null) {
                    val user = auth.currentUser!!
                    lifecycleScope.launch(Dispatchers.IO) {
                        userRepository.add(
                            User(
                                id = user.uid,
                                name = user.displayName!!,
                                profilePicture = user.photoUrl?.toString()
                            )
                        )
                    }
                    editor.putString(getString(R.string.account_firebase_uid), user.uid)
```
I think it's better to check if the user is not null before. The code becomes a bit simpler inside and we avoid adding null user to the database.